lms/djangoapps/monitoring/scripts/process_cookie_monitoring_logs.py

```diff
@@ -0,0 +1,207 @@
+
+
+# Note: Not all Open edX deployments will be affected by the same third-party cookies,
+#   but it is ok to have some of these cookies go unused.
+PARAMETERIZED_COOKIES = [
+    (re.compile(r"_gac_UA-(\d|-)+$"), "_gac_UA-{id}"),
+    (re.compile(r"_hjSession_\d+$"), "_hjSession_{id}"),
+    (re.compile(r"_hjSessionUser_\d+$"), "_hjSessionUser_{id}"),
+    (re.compile(r"ab.storage.deviceId.*$"), "ab.storage.deviceId.{id}"),
+    (re.compile(r"ab.storage.sessionId.*$"), "ab.storage.deviceId.{id}"),
+    (re.compile(r"ab.storage.userId.*$"), "ab.storage.userId.{id}"),
+    (re.compile(r"AMCV_\w+%40AdobeOrg$"), "AMCV_{id}@AdobeOrg"),
+    (re.compile(r"amplitude_id_.*$"), "amplitude_id_{id}"),
+    (re.compile(r"mp_\w+_mixpanel$"), "mp_{id}_mixpanel"),
+]
+
+
+@click.command()
+@click.option(
+    "--csv_input",
+    help="File name of .csv file with Splunk logs for large cookie headers.",
+    required=True
+)
+def main(csv_input):
+    """
+    Reads CSV of large cookie logs and processes and provides summary output.
+
+    Expected CSV format (from Splunk export searching on "BEGIN-COOKIE-SIZES"):
+
+        \b
+        _raw,_time,index,
+        ...
+
+    """
+    cookie_headers = _load_csv(csv_input)
+    processed_cookie_headers = process_cookie_headers(cookie_headers)
+    print_processed_cookies(processed_cookie_headers)
+
+
+def _load_csv(csv_file):
+    """
+    Reads CSV of large cookie data and returns a dict of details.
+
+    Arguments:
+        csv_file (string): File name for the csv
+
+    Returns a list of dicts containing parsed details for each cookie header log entry.
+
+    """
+    with open(csv_file) as file:
+        csv_data = file.read()
+    reader = csv.DictReader(csv_data.splitlines())
+
+    # Regex to match against log messages like the following:
+    #   BEGIN-COOKIE-SIZES(total=3773) user-info: 903, csrftoken: 64, ... END-COOKIE-SIZES
+    cookie_log_regex = re.compile(r"BEGIN-COOKIE-SIZES\(total=(?P<total>\d+)\)(?P<cookie_sizes>.*)END-COOKIE-SIZES")
+    cookie_sizes_strings_processed = set()
+    # Regex to match against just a single size, like the following:
+    #   csrftoken: 64
+    cookie_size_regex = re.compile(r"(?P<name>.*): (?P<size>\d+)")
+
+    cookie_headers = []
+    for row in reader:
+        cookie_header_sizes = {}
+
+        raw_cookie_log = row.get("_raw")
+        cookie_begin_count = raw_cookie_log.count("BEGIN-COOKIE-SIZES")
+        if cookie_begin_count == 0:
+            logging.info("No BEGIN-COOKIE-SIZES delimeter found. Skipping row.")
+        elif cookie_begin_count > 1:
+            # Note: this wouldn't parse correctly right now, and it isn't worth coding for.
+            logging.warning("Multiple cookie entries found in same row. Skipping row.")
+            continue
+        match = cookie_log_regex.search(raw_cookie_log)
+        if not match:
+            logging.error("Multiple cookie entries found in same row. Skipping row.")
```

Comment:
Duplicated message. Suggestion:
```suggestion
            logging.error("Malformed cookie entry. Skipping row.")
```

Reply:
Fixed.